src/core/patterns.js

```diff
@@ -59,3 +59,61 @@ foam.CLASS({
     }
   ]
 });
+
+/** Causes an class to pool its instances. create() will pull from the pool,
+ and destroy() will return instances to the pool. Object pools can be found
+ in <code>foam.__objectPools__</code>. */
+foam.CLASS({
+  package: 'foam.pattern',
+  name: 'Pooled',
+  axioms: [ foam.pattern.Singleton.create() ],
+
+  methods: [
+    function installInClass(cls) {
+      // Keeping the object pools in an accessible location allows them
+      // to be cleared out.
+      if ( ! foam.__objectPools__ ) { foam.__objectPools__ = {}; }
```

Comment:
The pool arrays would only contain destroyed objects that aren't being used. If you created a thousand things, destroyed them all and then moved on to something else, emptying the pools would free up those thousand retained things.